mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,404 @@
+        ###should note that current .mtg files contain channels for multiple data files###
+        ###going to move to have a single .mtg file per participant, condition, and montage###
+        ###combine coordinates and label our channels###
+        ###will label them based on ac, dc, and ph data###
+        boxy_coords = []
+        boxy_labels = []
+        data_types = ['AC','DC','Ph']
+        total_chans = detect_num*source_num
+        for i_type in data_types:
+            for i_coord in range(len(source_coords[0:total_chans])):
+                boxy_coords.append(np.mean(
+                    np.vstack((source_coords[i_coord], detect_coords[i_coord])),
+                    axis=0).tolist() + source_coords[i_coord] + 
+                    detect_coords[i_coord] + [chan_wavelength[i_coord]] + [0] + [0])
+                boxy_labels.append('S' + 
+                                       str(unique_source_labels.index(source_label[i_coord])+1)
+                                       + '_D' + 
+                                       str(unique_detect_labels.index(detect_label[i_coord])+1) 
+                                       + ' ' + chan_wavelength[i_coord] + ' ' + i_type)
+
+        ###montage only wants channel coords, so need to grab those, convert to###
+        ###array, then make a dict with labels###
+        for i_chan in range(len(boxy_coords)):
+            boxy_coords[i_chan] = np.asarray(boxy_coords[i_chan],dtype=np.float64) 
+            
+        for i_chan in range(len(all_coords)):
+            all_coords[i_chan] = np.asarray(all_coords[i_chan],dtype=np.float64) 
+
+        all_chan_dict = dict(zip(all_labels,all_coords))
+
+ 
+        ###make our montage###
+        montage_orig = make_dig_montage(ch_pos=all_chan_dict,coord_frame='head',
+                                                nasion = fiducial_coords[0],
+                                                lpa = fiducial_coords[1], 
+                                                rpa = fiducial_coords[2])
+        
+        ###for some reason make_dig_montage put our channels in a different order than what we input###
+        ###let's fix that. should be fine to just change coords and ch_names###
+        for i_chan in range(len(all_coords)):
+            montage_orig.dig[i_chan+3]['r'] = all_coords[i_chan]
+            montage_orig.ch_names[i_chan] = all_labels[i_chan]
+        
+        ###add an extra channel for our triggers for later###
+        boxy_labels.append('Markers')
+
+        info = create_info(boxy_labels,srate,ch_types='fnirs_raw')
+        info.update(dig=montage_orig.dig)
+
+        # Set up digitization
+        # These are all in MNI coordinates, so let's transform them to
+        # the Neuromag head coordinate frame
+        ###get our fiducials and transform matrix from fsaverage###
+        subjects_dir = op.dirname(mne.datasets.fetch_fsaverage())
+        fid_path = op.join(subjects_dir, 'fsaverage', 'bem', 'fsaverage-fiducials.fif')
```

Comment:
You don't need `fetch_fsaverage` for this, it's already in `mne/data`

https://github.com/mne-tools/mne-python/blob/master/mne/data/fsaverage/fsaverage-fiducials.fif